### PR TITLE
Iteration0 Project Base

### PR DESCRIPTION
**Pull request by aCard0s0**

### CODE REVIEW

File: xBrokerConnector/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# 1st Stage - Build
+FROM maven:3.6-openjdk-11 AS bluider
```
Comment (maiph) [inline xBrokerConnector/Dockerfile]: Probably that is what was wrong with it 🙂 
```suggestion
FROM maven:3.6-openjdk-11 AS builder
```

File: xBrokerConnector/pom.xml [hunk 2]
```diff
@@ -11,6 +11,104 @@
...
+            </plugin>
+        </plugins>
+    </build>
+
+    <parent>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.4.2</version>
+        <relativePath/> <!-- lookup parent from repository -->
+    </parent>
```
Comment (maiph) [inline xBrokerConnector/pom.xml:23]: Lets move this just before the `properties`

File: xBrokerConnector/src/main/java/dev/midas/xbc/XBCApp.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import dev.midas.xbc.config.ConfigManager;
+import dev.midas.xbc.stream.ListeningEngine;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class XBCApp implements CommandLineRunner{
```
Comment (maiph) [inline xBrokerConnector/src/main/java/dev/midas/xbc/XBCApp.java]:
```suggestion
public class XBCApp implements CommandLineRunner {
```

File: xBrokerConnector/src/main/java/dev/midas/xbc/XBCApp.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Autowired
+    private final ListeningEngine listening;
+
+    public XBCApp(ListeningEngine listening) {
+        this.listening = listening;
+    }
```
Comment (maiph) [inline xBrokerConnector/src/main/java/dev/midas/xbc/XBCApp.java]: I don't think we need the constructor if we use the `@Autowired`, but may be wrong

File: xBrokerConnector/src/main/java/dev/midas/xbc/config/ConfigManager.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    public ConfigManager() {}
+
+    public ExchangeConfig[] loadExchangeConfig() {
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        mapper.findAndRegisterModules();
+        ExchangeConfig[] configs = null;
+
+        try {
+            configs = mapper.readValue(new File("xBrokerConnector/src/main/resources/exchange.yaml"), ExchangeConfig[].class);
```
Comment (maiph) [inline xBrokerConnector/src/main/java/dev/midas/xbc/config/ConfigManager.java]: Loading files like this only works if you have the source code locally, we need to load resources from the classpath

File: xBrokerConnector/src/main/java/dev/midas/xbc/service/ExchangeService.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    public ExchangeService(ListeningEngine engine) {
+        this.engine = engine;
+    }
+
+    public void stopAllListeners() {
+        this.engine.stopAllListeners();
+    }
+
+    public void startListener(String exchangeId, String marketId) {
+        // Deserilaztion
```
Comment (maiph) [inline xBrokerConnector/src/main/java/dev/midas/xbc/service/ExchangeService.java:27]:
```suggestion
        // Deserialization
```

File: xBrokerConnector/src/main/java/dev/midas/xbc/service/ExchangeService.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        this.engine.startListener(KrakenStreamingExchange.class, CurrencyPair.BTC_EUR);
+    }
+
+    public void stopListener(String exchangeId, String marketId) {
+        // Deserilaztion
```
Comment (maiph) [inline xBrokerConnector/src/main/java/dev/midas/xbc/service/ExchangeService.java:32]:
```suggestion
        // Deserialization
```

File: xBrokerConnector/src/main/java/dev/midas/xbc/stream/DisposableFactory.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ *  This factory currently creates and destroys markets streams.
+ */
+@Component
+public class DisposableFactory {
+
+    private static final Logger LOG = LogManager.getLogger();
```
Comment (maiph) [inline xBrokerConnector/src/main/java/dev/midas/xbc/stream/DisposableFactory.java]:
```suggestion
    private static final Logger LOG = LogManager.getLogger(DisaposableFactory.class.getName());
```

File: xBrokerConnector/src/main/java/dev/midas/xbc/stream/DisposableFactory.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ *  This factory currently creates and destroys markets streams.
+ */
+@Component
+public class DisposableFactory {
```
Comment (maiph) [inline xBrokerConnector/src/main/java/dev/midas/xbc/stream/DisposableFactory.java]: `DisposableFactory` is more than just a factory and needs a better name 😛

File: xBrokerConnector/src/main/java/dev/midas/xbc/XBCApp.java (hunk 4 above)
Comment (aCard0s0) [reply xBrokerConnector/src/main/java/dev/midas/xbc/XBCApp.java]: you are right! thanks

File: xBrokerConnector/src/main/java/dev/midas/xbc/config/ConfigManager.java (hunk 5 above)
Comment (aCard0s0) [reply xBrokerConnector/src/main/java/dev/midas/xbc/config/ConfigManager.java]: Thank you to bring that to my attention ;)

File: xBrokerConnector/src/main/java/dev/midas/xbc/stream/DisposableFactory.java (hunk 9 above)
Comment (aCard0s0) [reply xBrokerConnector/src/main/java/dev/midas/xbc/stream/DisposableFactory.java]: I totally agree.

**Review by aCard0s0 — COMMENTED** [review]
Thank you for the review!

**Pull request merged**